### PR TITLE
249711 Fix bug that guest doesn't get a feedback message after submitting the form. 

### PR DESCRIPTION
**Pull request by tgietzen**
Problem was that the participant constructor had too few arguments. Clean up MealController and psalm baseline.

### CODE REVIEW

File: src/Mealz/MealBundle/Controller/MealController.php [hunk 1]
```diff
@@ -156,57 +154,22 @@ private function getMealErrorCode($meal)
...
-     * @return JsonResponse Ajax-Request
-     */
-    private function generateDeleteResponse($meal, $participant)
+    private function generateResponse(string $route, string $action, Meal $meal, Participant $participant): JsonResponse
     {
-        $ajaxResponse = new JsonResponse();
-        $ajaxResponse->setData(
+        $response = new JsonResponse();
+        $response->setData(
             array(
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/MealController.php:161]: We can use the short array syntax here.

File: src/Mealz/MealBundle/Controller/MealController.php [hunk 2]
```diff
@@ -266,7 +226,7 @@ private function sendMealTakenMattermostMsg(string $dishTitle, int $remainingOff
      * Marks meals that are being offered.
      * @return JsonResponse
      */
-    public function updateOffers()
+    public function updateOffers(): JsonResponse
     {
         $mealsArray = array();
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/MealController.php:231]: Same as above.

File: src/Mealz/MealBundle/Tests/Controller/MealControllerTest.php [hunk 3]
```diff
@@ -339,13 +337,7 @@ public function testNewMealFlag(): void
...
     }
 
-    /**
-     * Gets all the participants for a meal.
-     *
-     * @param Meal $meal Meal instance
-     * @return array
-     */
-    private function getMealParticipants($meal)
+    private function getMealParticipants(Meal $meal): array
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Tests/Controller/MealControllerTest.php:340]: We can add "@return Participant[]" to PHP-Doc. It conveys more information than just simple array.

File: src/Mealz/MealBundle/Controller/ParticipantController.php [hunk 4]
```diff
@@ -232,4 +208,19 @@ public function editParticipation(Week $week)
             'prototype' => $prototype,
         ));
     }
+
+    private function generateResponse(string $route, string $action, Participant $participant): JsonResponse
+    {
+        $response = new JsonResponse();
+        $response->setData(
+            array(
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/ParticipantController.php:216]: Short array syntax would be nice.

File: src/Mealz/MealBundle/Controller/ParticipantController.php [hunk 5]
```diff
@@ -232,4 +208,19 @@ public function editParticipation(Week $week)
             'prototype' => $prototype,
         ));
     }
+
+    private function generateResponse(string $route, string $action, Participant $participant): JsonResponse
+    {
+        $response = new JsonResponse();
+        $response->setData(
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/ParticipantController.php:215]: I believe data can be specified as the first constructor parameter to JsonResponse. We can save an extra method call this way. May be something like "return new JsonResponse([...]);".

File: src/Mealz/MealBundle/Controller/ParticipantController.php [hunk 6]
```diff
@@ -188,11 +165,10 @@ public function list(DayRepository $dayRepo): Response
     }
 
     /**
-     * edit participation
      * @param Week $week
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/ParticipantController.php:168]: Redundant @param and @return annotations.

File: src/Mealz/MealBundle/Controller/ParticipantController.php [hunk 7]
```diff
@@ -136,25 +125,15 @@ public function swap(Participant $participant, NotifierInterface $notifier)
...
-            'id' => $participant->getId(),
-            'actionText' => $this->get('translator')->trans('swapped', array(), 'action'),
-        ));
-
-        return $ajaxResponse;
+        return $this->generateResponse('MealzMealBundle_Participant_unswap', 'swapped', $participant);
     }
 
     /**
      * @param Participant $participant
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/ParticipantController.php:132]: Redundant @param and @return annotation.

File: src/Mealz/MealBundle/Controller/ParticipantController.php [hunk 8]
```diff
@@ -71,9 +67,10 @@ public function delete(Participant $participant)
      * Offers an existing participation by setting the participant's 'offeredAt' value to the timestamp.
      * Takes an existing offer back by setting the 'offeredAt' value back to 0.
      * @param Participant $participant
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/ParticipantController.php:69]: @ annotations can be removed.

File: src/Mealz/MealBundle/Controller/MealController.php [hunk 9]
```diff
@@ -492,7 +424,7 @@ private function createEmptyNonPersistentWeek(DateTime $dateTime): Week
      * @param Meal $meal
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/MealController.php:424]: Annotations can be removed.

File: src/Mealz/MealBundle/Controller/MealController.php [hunk 10]
```diff
@@ -378,39 +325,31 @@ protected function renderGuestForm($form)
...
 
     /**
-     * Adds a participation for every chosen meal.
-     *
-     * @param Meal              $meals           The meals
-     * @param Profile           $profile         The profile
-     * @param MealRepository    $mealRepository  The meal repository
-     * @param translator        $translator      The Translator Object
-     *
+     * @param array $meals
```
Comment (chetan-thapliyal) [inline src/Mealz/MealBundle/Controller/MealController.php:344]: @param annotation is redundant.

File: src/Mealz/MealBundle/Controller/ParticipantController.php (hunk 8 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/ParticipantController.php:69]: Done.

File: src/Mealz/MealBundle/Controller/ParticipantController.php (hunk 7 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/ParticipantController.php:132]: Done.

File: src/Mealz/MealBundle/Controller/ParticipantController.php (hunk 6 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/ParticipantController.php:168]: Removed.

File: src/Mealz/MealBundle/Controller/ParticipantController.php (hunk 5 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/ParticipantController.php:215]: Done.

File: src/Mealz/MealBundle/Controller/ParticipantController.php (hunk 4 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/ParticipantController.php:216]: Done.

File: src/Mealz/MealBundle/Tests/Controller/MealControllerTest.php (hunk 3 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Tests/Controller/MealControllerTest.php:340]: Ok.

File: src/Mealz/MealBundle/Controller/MealController.php (hunk 1 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/MealController.php:161]: Done.

File: src/Mealz/MealBundle/Controller/MealController.php (hunk 2 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/MealController.php:231]: Done.

File: src/Mealz/MealBundle/Controller/MealController.php (hunk 10 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/MealController.php:344]: Removed.

File: src/Mealz/MealBundle/Controller/MealController.php (hunk 9 above)
Comment (tgietzen) [reply src/Mealz/MealBundle/Controller/MealController.php:424]: Done.

**Review by sourcesoldier — APPROVED** [review]

**Pull request merged**